pkg/meta/sql.go

```diff
@@ -893,7 +898,40 @@ func (m *dbMeta) roTxn(ctx context.Context, f func(s *xorm.Session) error) error
 		}
 		return err
 	}
-	logger.Warnf("Already tried 50 times, returning: %s", lastErr)
+	logger.Warnf("Already tried %d times, returning: %s", maxRetry, lastErr)
+	return lastErr
+}
+
+func (m *dbMeta) simpleTxn(ctx context.Context, f func(s *xorm.Session) error) error {
+	start := time.Now()
+	defer func() { m.txDist.Observe(time.Since(start).Seconds()) }()
+	s := m.db.NewSession()
```

Comment:
Since there are no transactions, perhaps `NewSession` does not need to be used every time, just use one globally or `xorm.Engine` direct

Reply:
we use Session as an interface, but it's not thread-safe, so we can use a pool of sessions